Sources/_CryptoExtras/Util/BoringSSLHelpers.swift

```diff
@@ -86,7 +86,11 @@ extension FixedWidthInteger {
             CCryptoBoringSSL_BN_clear(&bn)
         }
 
+        #if arch(arm) || arch(i386)
+        CCryptoBoringSSL_BN_set_word(&bn, UInt32(self))
+        #else
         CCryptoBoringSSL_BN_set_word(&bn, UInt64(self))
+        #endif
```

Comment:
Thanks for this patch! Instead of the conditional compilation, can we use `.init(self)` instead and let the type inference engine figure it out?

Reply:
Relatedly, the `precondition` on line 81 is clearly wrong in the 32-bit environment. Can we flip that over to `precondition(self.bitWidth <= UInt.bitWidth)`?

Reply:
Done, I cross-compiled this new patch for Android armv7 without a problem.